web/src/components/Gov2GoModal.tsx

```diff
@@ -0,0 +1,199 @@
+  });
+
+  const hasErrors = (errors?: Partial<ProfileFieldErrorMap>) =>
+    fields.some((i) => (errors ?? fieldStates)[i]?.invalid == true);
+
+  const onValidation = (field: ProfileFields, invalid: boolean) => {
+    setFieldStates({ ...fieldStates, [field]: { ...fieldStates[field], invalid } });
+  };
+
+  const onSubmit = async () => {
+    if (!userData) return;
+    setOnSubmitClicked(true);
+    const errors = getErrors();
+    setFieldStates((prev) => ({ ...prev, ...errors }));
+    if (hasErrors(errors)) return;
+    setIsLoading(true);
+    let userDataToSet: UserData;
+    try {
+      userDataToSet = await postTaxRegistrationOnboarding({
+        taxId: profileData.taxId as string,
+        businessName: profileData.businessName,
+      });
+    } catch {
+      setOnAPIfailed("UNKNOWN");
+      setIsLoading(false);
+      return;
+    }
+    userDataToSet = {
+      ...userDataToSet,
+      profileData: { ...userDataToSet.profileData, taxId: profileData.taxId },
+    };
+
+    await update(userDataToSet);
+    if (userDataToSet.taxFilingData.state == "SUCCESS") {
+      setIsLoading(false);
+      props.onSuccess();
+    }
+
+    if (userDataToSet.taxFilingData.state == "PENDING") {
+      setIsLoading(false);
+      props.close();
+    }
+
+    if (userDataToSet.taxFilingData.state == "FAILED") {
+      setFieldStates((prev) => ({
+        ...fields.reduce(
+          (p, c) => {
+            p[c] = { ...p[c], invalid: true };
+            return p;
+          },
+          { ...prev }
+        ),
+      }));
+      setOnAPIfailed("FAILED");
+      setIsLoading(false);
+    }
+
+    if (userDataToSet.taxFilingData.state == "API_ERROR") {
+      setOnAPIfailed("UNKNOWN");
+      setIsLoading(false);
+    }
+  };
+
+  useMountEffectWhenDefined(() => {
+    if (userData) {
+      setProfileData(userData.profileData);
+    }
+  }, userData);
+
+  const onClose = () => {
+    props.close();
+    userData && setProfileData(userData.profileData);
+    setOnAPIfailed(undefined);
+    setOnSubmitClicked(false);
+    setFieldStates(createProfileFieldErrorMap());
+  };
+
+  return (
+    <ProfileDataContext.Provider
+      value={{
+        state: {
+          profileData: profileData,
+          flow: "OWNING",
+          municipalities: [],
+        },
+        setUser: () => {},
+        setProfileData,
+        onBack: () => {},
+      }}
+    >
+      <Backdrop sx={{ zIndex: 20000 }} open={isLoading}>
+        <CircularProgress aria-label="Loading indicator" aria-busy={true} />
+      </Backdrop>
+      <ModalTwoButton
+        isOpen={props.isOpen}
+        close={onClose}
+        title={Config.dashboardDefaults.taxCalendarModalHeader}
+        primaryButtonText={Config.dashboardDefaults.taxCalendarModalNextButton}
+        primaryButtonOnClick={onSubmit}
+        secondaryButtonText={Config.dashboardDefaults.taxCalendarModalPreviousButton}
+      >
+        {hasErrors() && onSubmitClicked && !apiFailed ? (
```

Comment:
we prefer to use `&&` syntax in the jsx instead of a ternary with an unnecessary `: ( <> </> )}` at the end, which just takes up space and makes things harder to read

Reply:
same comment below